kyverno-policy-deployment/sharing-policies.md

```diff
@@ -0,0 +1,82 @@
+# Sharing Kyverno Policies with Customers
+
+This mini-RFC documents the result of [a discussion](https://github.com/giantswarm/giantswarm/issues/26365) about Kyverno policies we deploy in customer clusters.
+
+The outcome of this RFC should be to clarify the general expectation of Kyverno policies in workload clusters, and optionally to generate additional discussions for the points that remain unclear.
+
+## TL;DR
+
+We used a Kyverno policy to implement some behavior, and didn't anticipate that customers might also use the policy. When we changed the policy for our usage, it broke the functionality for the customer.
+
+Since we will give customers lots of Kyverno policies for Pod Security Standards (PSS) in the future, we should let them assume that policies we deploy are intended for their usage.
+
+If that isn't true for a certain policy, the responsibility is on us to make that clear and reduce the likelihood that customers will use it mistakenly.
+
+## Context
+
+To support running a cluster behind a proxy, we created a Kyverno mutating policy which injects environment variables containing proxy details into Pod resources at admission time.
+Pods are mutated by this policy only if they have a certain label.
+
+A customer noticed this capability, and began using the label to have their pods also mutated to include the proxy information.
+
+We did not realize that the customer had begun using it, and modified the policy in a way that made it no longer work for the customer.
+
+The resulting discussion is about what expectations we and customers can have about the stability of Kyverno policies in workload clusters.
+
+## Discussion Points
+
+In the resulting issue and subsequent sync, the following questions arose:
+
+1. Is a Kyverno policy the correct place to implement this logic to begin with? Is this a permanent solution or is there an alternative to be implemented in the long term? --> Out of scope of this RFC.
+1. What is the expectation of the proxy support in our product? Do we provide a mechanism for customer workloads, or must customers configure their own workloads to use the proxy? --> Out of the scope of this RFC.
+1. To what extent are Kyverno policies part of our public interface to customers? --> Focus of this RFC.
+
+The first two questions are about the specific policy in question and are worth separate discussions, but not in this RFC, which hopes to address the general case of Kyverno policies in WCs.
+
+## Current State
+
+We deploy several types of Kyverno policies to management (MC) and/or workload clusters (WC):
+
+- Pod Security Standards (PSS) policies --> MC + WC
+- UX policies (e.g. block deletion of Orgs which still have clusters) --> MC
+- DX policies (e.g. limit Crossplane or External Secrets management) --> MC
+- Observability policies (e.g. ServiceMonitor defaulting) --> MC (+ WC?)
+- Connectivity policy for proxy --> MC + WC
```

Comment:
Actually, we only deploy the proxy policy on MCs. On this case, the customer was deploying stuff into the MC.

Reply:
Oh I see. I thought the policy was also intended for WC components